src/AllowlistHook.sol

```diff
@@ -120,27 +122,53 @@ contract AllowlistHook is BaseTickRangeHook, AllowlistHookStorageLayout {
     /**
      * @dev    Hook that is called before a swap is executed.
      * @dev    Will revert if the sender is not allowed to swap.
-     * @param  sender_ The address of the sender initiating the swap (i.e. most commonly the Swap Router).
+     * @param  sender_   The address of the sender initiating the swap (i.e. most commonly the Swap Router).
+     * @param  key_      Underlying pool configuration information.
+     * @param  params_   Swap parameters including direction and amount.
+     * @param  hookData_ Encoded authorization message from Predicate.
      * @return A tuple containing the selector of this function, the delta for the swap, and the LP fee.
      */
     function _beforeSwap(
         address sender_,
-        PoolKey calldata /* poolKey */,
-        IPoolManager.SwapParams calldata /* params */,
-        bytes calldata /* hookData */
+        PoolKey calldata key_,
+        IPoolManager.SwapParams calldata params_,
+        bytes calldata hookData_
     ) internal override returns (bytes4, BeforeSwapDelta, uint24) {
-        AllowlistHookStorage storage $ = _getAllowlistHookStorage();
-
-        if ($.isSwappersAllowlistEnabled) {
-            if (!$.swapRouters[sender_]) {
+        if (isSwappersAllowlistEnabled) {
```

Comment:
few questions here 
- should swap router check still be enabled even if `isSwappersAllowlistEnabled` is false. Swap should not be coming from unknown router 
- I wonder if we ever will need situation where whitelist id disabled, but predicate check is enabled. 

In this case it is better to have all 3 checks: router, whitelisted address, predicate check separately